### PR TITLE
feat: add location service icon which can be used with maplibre-gl-geocoder

### PR DESCRIPTION
**Pull request by thaddmt**
#### Description of changes
- Added default svg icon used by location services
- Added utility function to create an HTMLImageElement that uses that svg icon, can be passed to maplibre-gl-geocoder
- Added utility function createAmplifyGeocoder that sets the marker by default
- Added documentation

Screen Shot 2021-08-17 at 2 49 15 PM

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Relevant documentation is changed or added (and PR referenced)

### CODE REVIEW

File: src/custom.d.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+declare module "*.svg" {
+  const content: any;
+  export default content;
+}
```
Comment (Amplifiyer) [inline src/custom.d.ts:4]: what's the purpose of this file?

File: src/createMarker.ts [hunk 2]
```diff
@@ -32,3 +33,9 @@ export function createMarker(options?: CreateMarkerOptions): ImageMarker {
     },
   };
 }
+
+export function createDefaultIcon(): HTMLImageElement {
+  const customIcon = new Image(24, 24);
+  customIcon.src = icon;
+  return customIcon;
+}
```
Comment (Amplifiyer) [inline src/createMarker.ts:41]: Probably should be it's own file. the following looks weird.
```
import { createDefaultIcon } from 'createMarker`
```

File: API.md [hunk 3]
```diff
@@ -39,66 +40,66 @@ DrawPoints utility function for adding points to a map based on coordinate data
...
+- `clusterSymbolLayerId` **[String][8]** The [layer][20] used for creating styling the number that shows the count of points in a cluster
+- `unclusteredLayerId` **[String][8]** The [layer][21] used for creating and styling the individual points on the map and the popup when clicking on a point
 
 ## AmplifyGeocoderAPI
 
 An object wrapping Amplify Geo search APIs and returns `forwardGeocode` and `reverseGeocode` methods which are used by [maplibre-gl-geocoder][22] to perform search
 
+## createDefaultIcon
+
+A method that returns an Icon in the form of an [HTMLImageElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement). That can be passed to the marker element in [maplibre-gl-geocoder][22]. Example:
```
Comment (Amplifiyer) [inline API.md:69]:
```suggestion
A method that returns an Icon in the form of an [HTMLImageElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement) that can be passed to the marker element in [maplibre-gl-geocoder][22]. Example:
```

File: src/AmplifyMapLibreGeocoder.ts [hunk 4]
```diff
@@ -1,4 +1,6 @@
 import { Geo } from "@aws-amplify/geo";
+import MaplibreGeocoder from "@maplibre/maplibre-gl-geocoder";
+import { createDefaultIcon } from "./createMarker";
```
Comment (Amplifiyer) [inline src/AmplifyMapLibreGeocoder.ts:3]: Would it work if we just load/import the geocoder's css here such that developers don't have to in their app?
`import "@maplibre/maplibre-gl-geocoder/dist/maplibre-gl-geocoder.css";`

File: src/custom.d.ts (hunk 1 above)
Comment (thaddmt) [reply src/custom.d.ts:4]: It's to allow importing svg's in ts files

File: src/AmplifyMapLibreGeocoder.ts (hunk 4 above)
Comment (thaddmt) [reply src/AmplifyMapLibreGeocoder.ts:3]: Will try it out and report back!

Comment (thaddmt) [reply src/AmplifyMapLibreGeocoder.ts]: It works!

Comment (Amplifiyer) [reply src/AmplifyMapLibreGeocoder.ts]: Awesome thanks, were you able to test both a react app and static html pages?

Comment (thaddmt) [reply src/AmplifyMapLibreGeocoder.ts]: I tested in a react app but not in a static html page

Comment (Amplifiyer) [reply src/AmplifyMapLibreGeocoder.ts]: We do need to test both scenarios. Last time there was a regression in static html pages.

**Review by Milan-Shah — APPROVED** [review]
LGTM!

File: src/AmplifyMapLibreGeocoder.ts (hunk 4 above)
Comment (TreTuna) [reply src/AmplifyMapLibreGeocoder.ts]: Yeah, as long as this works in the static pages, the code looks good to me.

**Review by Amplifiyer — APPROVED** [review]

**Pull request merged**